cmd/common.go

```diff
@@ -348,78 +349,32 @@ func Lookup(c *cliclient.MasterClient, name string, types ...string) (*ntypes.Re
 	var byName *ntypes.Resource
 
 	for _, schemaType := range types {
-		rt, err := GetResourceType(c, schemaType)
-		if err != nil {
-			logrus.Debugf("Error GetResourceType: %v", err)
-			return nil, err
-		}
-		var schemaClient clientbase.APIBaseClientInterface
-		// the schemaType dictates which client we need to use
-		if c.CAPIClient != nil {
-			if strings.HasPrefix(rt, "cluster.x-k8s.io") {
-				schemaClient = c.CAPIClient
-			}
-		}
-		if c.ManagementClient != nil {
-			if _, ok := c.ManagementClient.APIBaseClient.Types[rt]; ok {
-				schemaClient = c.ManagementClient
-			}
-		}
-		if c.ProjectClient != nil {
-			if _, ok := c.ProjectClient.APIBaseClient.Types[rt]; ok {
-				schemaClient = c.ProjectClient
-			}
-		}
-		if c.ClusterClient != nil {
-			if _, ok := c.ClusterClient.APIBaseClient.Types[rt]; ok {
-				schemaClient = c.ClusterClient
-			}
-		}
-
 		// Attempt to get the resource by ID
-		var resource ntypes.Resource
-
-		if err := schemaClient.ByID(schemaType, name, &resource); !clientbase.IsNotFound(err) && err != nil {
-			logrus.Debugf("Error schemaClient.ByID: %v", err)
+		resource, err := LookupByID(c, name, schemaType)
+		if err != nil && !strings.Contains(err.Error(), "Not found") {
 			return nil, err
-		} else if err == nil && resource.ID == name {
-			return &resource, nil
 		}
-
-		// Resource was not found assuming the ID, check if it's the name of a resource
-		var collection ntypes.ResourceCollection
-
-		listOpts := &ntypes.ListOpts{
-			Filters: map[string]interface{}{
-				"name":         name,
-				"removed_null": 1,
-			},
+		if resource != nil {
+			return resource, nil
 		}
 
-		if err := schemaClient.List(schemaType, listOpts, &collection); !clientbase.IsNotFound(err) && err != nil {
-			logrus.Debugf("Error schemaClient.List: %v", err)
+		// Resource was not found assuming the ID, check if it's the name of a resource
+		resource, err = LookupByName(c, name, schemaType)
+		if err != nil && !strings.Contains(err.Error(), "Not found") {
```

Comment:
We should always try and avoid doing string checks on `err.Error()` and instead do a type check. 
To accomplish this, we will need to:
-  define `errNotFound` as a package variable.
- Return `errNotFound` around line 436.
- Check if `err == errNotFound`